Project/core/src/test/java/Model/GameTests.java

```diff
@@ -0,0 +1,132 @@
+        assertEquals(1, game.getCurrentTime(), "Initial game time should be 1.");
+        assertEquals(0, game.getAmountRecruited(), "Amount recruited should start at 0.");
+        assertEquals(0, game.getMindSlipCount(), "Mind slip count should start at 0.");
+        assertEquals(board, game.getBoard(), "Board should match the one passed in.");
+        assertEquals(2, game.getPlayers().size(), "Game should have 2 players.");
+    }
+
+    @Test
+    void testSetGameOver() {
+        game.setGameOver();
+        assertTrue(game.isGameOver(), "Game should be marked as over.");
+        assertThrows(IllegalStateException.class, () -> game.setGameOver(),
+        "Constructor should throw an exception if game is already over");
+    }
+
+    @Test
+    void testSetCurrentPlayer() {
+        game.setCurrentPlayer(player2);
+        assertEquals(player2, game.getCurrentPlayer(), "Current player should be Player 2.");
+        game.setCurrentPlayer(player1);
+        assertEquals(player1, game.getCurrentPlayer(), "Current player should be Player 1.");
+        assertThrows(IllegalArgumentException.class, () -> game.setCurrentPlayer(new RougeAgent(2, null)),
+            "Setting the next player to one not in the game should throw IllegalArgumentException.");
+        assertThrows(IllegalArgumentException.class, () -> game.setCurrentPlayer(null),
+            "Setting the next player to null should throw an IllegalArgumentException.");
+    }
+
+    @Test
+    void testIncrementTime() {
+        game.incrementTime();
+        assertEquals(2, game.getCurrentTime(), "Game time should increment to 2.");
+        game.incrementTime();
+        assertEquals(3, game.getCurrentTime(), "Game time should increment to 3.");
+    }
+
+    @Test
+    void testAddAmountRecruited() {
+        game.addAmountRecruited(1);
+        assertEquals(1, game.getAmountRecruited(), "Amount recruited should increase to 1.");
+        game.addAmountRecruited(2);
+        assertEquals(3, game.getAmountRecruited(), "Amount recruited should increase to 3.");
+        assertThrows(IllegalArgumentException.class, () -> game.addAmountRecruited(-1),
+        "Adding a negative number to the amount recruited should throw an IllegalArgumentException.");
+        assertThrows(IllegalArgumentException.class, () -> game.addAmountRecruited(0),
+        "Adding zero to the amount recruited should throw an IllegalArgumentException.");
+    }
+
+    @Test
+    void testIncrementMindSlipCount() {
+        game.incrementMindSlipCount();
+        assertEquals(1, game.getMindSlipCount(), "Mind slip count should increase to 1.");
+        game.incrementMindSlipCount();
+        assertEquals(2, game.getMindSlipCount(), "Mind slip count should increase to 2.");
+        assertThrows(IllegalStateException.class, () -> game.incrementMindSlipCount(),
+        "Increaseing the mind slip count above 2 should throw an IllegalStateException.");
```

Comment:
Regarding my earlier comment about mindslips this is correct! You can never have above two.